### ISSUE TITLE
CompatHelper fails with `LoadError: type Nothing has no field fork`

### ISSUE THREAD

**Issue by devmotion** [JuliaRegistries/CompatHelper.jl#391]
CompatHelper in the Turing repo fails with [the following error](https://github.com/TuringLang/Turing.jl/runs/3452016425?check_suite_focus=true#step:4:43):
```julia
ERROR: LoadError: type Nothing has no field fork
Stacktrace:
  [1] getproperty(x::Nothing, f::Symbol)
    @ Base ./Base.jl:33
  [2] not_pr_fork(repo::GitForge.GitHub.Repo, pr::GitForge.GitHub.PullRequest)
    @ CompatHelper ~/.julia/packages/CompatHelper/PrHrN/src/pull_requests.jl:38
```
The action fails also with the recommended Github action: https://github.com/TuringLang/Turing.jl/pull/1692

**Comment by DilumAluthge** [comment]
@mattBrzezinski @fchorney

**Comment by DilumAluthge** [comment]
I think this might be a bug in GitForge.jl, but I'm not 100% sure.

@christopher-dG

**Comment by DilumAluthge** [comment]
On the CompatHelper.jl side, probably the easiest is to assume that if `repo.fork` is `nothing`, we just assume that the repo is not a fork?

**Comment by DilumAluthge** [comment]
Yeah, it seems that the source of this error is that there is an open PR (https://github.com/TuringLang/Turing.jl/pull/1380) on Turing.jl that is made from a fork, and that fork has been deleted, but the PR is still open.

We should figure out how to handle this more gracefully. I don't know if that logic needs to go in CompatHelper or in GitForge.jl.

For now, I've made a quick "band-aid" PR just so that CompatHelper stops crashing on this case: #392

But we need a better long-term solution, I think.

**Comment by DilumAluthge** [comment]
Thanks for reporting, @devmotion!

Can you trigger a new run using a workflow dispatch event? I just released 3.0.8, which hopefully will not crash on this case.

**Comment by devmotion** [comment]
It works now: https://github.com/TuringLang/Turing.jl/runs/3452227042?check_suite_focus=true :tada: 

Thanks a lot for the bugfix release and how promptly you addressed the issue :heart:
